File: server/src/main/java/ru/onlinewallet/entity/account/AccountBill.java
package ru.onlinewallet.entity.account;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.Instant;

@Data
@Entity
@NoArgsConstructor
@Table(name = "w_account_bills")
@PersistenceUnit(unitName = "postgres")
public class AccountBill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "account_id", nullable = false)
    private Long accountId;

    @ManyToOne
    @JoinColumn(name = "currency_id")
    private Currency currency;

    @Column(name = "start_balance")
    private Double startBalance;

    @Column(name = "balance")
    private Double balance;

    @ManyToOne
    @JoinColumn(name = "account_id", referencedColumnName = "id", insertable = false, updatable = false)
    private Account account;

    @Column
    private double rate;

    @Column(name = "maturity_date")
    private Instant maturityDate;

}

File: frontend/src/app/entities/account-bill.ts
import {Currency} from "./currency";

export class AccountBill {
  public id: number;
  public accountId: number;
  public currency: Currency;
  public startBalance: number;
  public balance: number;
  public rate: number;
  public maturityDate: string;

  constructor(json: any) {
    this.id = json.id;
    this.accountId = json.accountId;
    this.currency = new Currency(json.currency);
    this.startBalance = json.startBalance;
    this.balance = json.balance;
    this.rate = json.rate;
    this.maturityDate = json.maturityDate;
  }
}
